pkgs/by-name/mi/mihomo/package.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, fetchFromGitHub
+, buildGoModule
+}:
+
+buildGoModule rec {
+  pname = "mihomo";
+  version = "1.18.0";
+
+  src = fetchFromGitHub {
+    owner = "MetaCubeX";
+    repo = "mihomo";
+    rev = "v${version}";
+    hash = "sha256-lxiPrFPOPNppxdm2Ns4jaMHMRCYFlMz2h2rf7x0gv9c=";
+  };
+
+  vendorHash = "sha256-b7q0e3HHolVhwNJF0kwvwuVy8ndJLc0ITMl+0/YtSjA=";
+
+  excludedPackages = [ "./test" ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/metacubex/mihomo/constant.Version=${version}"
+  ];
+
+  tags = [
+    "with_gvisor"
+  ];
+
+  # network required
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A simple Python Pydantic model for Honkai: Star Rail; former clash-meta";
```

Comment:
Mihomo's **main** branch is just to avoid China government's censorship, really main branch is [Alpha Branch](https://github.com/MetaCubeX/mihomo/tree/Alpha), so description would better be "Another Clash Kernel." or others?